components/match2/wikis/clashroyale/match_group_input_custom.lua

```diff
@@ -0,0 +1,723 @@
+				map.winner = scoreIndex
+			else
+				break
+			end
+		end
+	else
+		local winnerInput = tonumber(map.winner)
+		if String.isNotEmpty(map.walkover) then
+			local walkoverInput = tonumber(map.walkover)
+			if walkoverInput == 1 then
+				map.winner = 1
+			elseif walkoverInput == 2 then
+				map.winner = 2
+			elseif walkoverInput == 0 then
+				map.winner = 0
+			end
+			map.walkover = Table.includes(ALLOWED_STATUSES, map.walkover) and map.walkover or 'L'
+			map.scores = {-1, -1}
+			map.resulttype = 'default'
+		elseif map.winner == 'skip' then
+			map.scores = {0, 0}
+			map.scores = {-1, -1}
+			map.resulttype = 'np'
+		elseif winnerInput == 1 then
+			map.scores = {1, 0}
+		elseif winnerInput == 2 then
+			map.scores = {0, 1}
+		end
+	end
+
+	return map
+end
+
+function CustomMatchGroupInput._processPlayerMapData(map, match)
+	local participants = {}
+	local submatchOpponentPlayerNumbers = {}
+
+	for opponentIndex = 1, MAX_NUM_OPPONENTS do
+		local opponent = match['opponent' .. opponentIndex]
+
+		if opponent.type == Opponent.team then
+			local playerNumber = CustomMatchGroupInput._processTeamPlayerMapData(
+				opponent.match2players or {},
+				opponentIndex,
+				map,
+				participants
+			)
+
+			table.insert(submatchOpponentPlayerNumbers, playerNumber)
+		elseif opponent.type == Opponent.literal then
+			table.insert(submatchOpponentPlayerNumbers, opponent.type)
+		else
+			table.insert(submatchOpponentPlayerNumbers, #opponent.match2players)
+
+			CustomMatchGroupInput._processDefaultPlayerMapData(
+				opponent.match2players or {},
+				opponentIndex,
+				map,
+				participants
+			)
+		end
+	end
+
+	map.mode = table.concat(submatchOpponentPlayerNumbers, 'v')
+	map.participants = participants
+
+	return map
+end
+
+function CustomMatchGroupInput._processDefaultPlayerMapData(players, opponentIndex, map, participants)
+	for playerIndex = 1, #players do
+		participants[opponentIndex .. '_' .. playerIndex] = {
+			played = true,
+			cards = CustomMatchGroupInput._readCards(map['t' .. opponentIndex .. 'p' .. playerIndex .. 'c']),
+		}
+	end
+end
+
+function CustomMatchGroupInput._processTeamPlayerMapData(players, opponentIndex, map, participants)
+	local tbdIndex = 0
+	local appendIndex = #players + 1
+
+	local playerIndex = 1
+	local playerKey = 't' .. opponentIndex .. 'p' .. playerIndex
+	while playerIndex <= MAX_NUM_PLAYERS_PER_MAP and (String.isNotEmpty(map[playerKey]) or
+		String.isNotEmpty(map[playerKey .. 'link']) or String.isNotEmpty(map[playerKey .. 'c'])) do
+
+		local player = map[playerKey .. 'link'] or map[playerKey]
+		if String.isEmpty(player) or Table.includes(TBD_PLAYERS, player:lower()) then
+			tbdIndex = tbdIndex + 1
+			player = TBD .. tbdIndex
+		else
+			-- allows fetching the link of the player from preset wiki vars
+			player = mw.ext.TeamLiquidIntegration.resolve_redirect(
+				map[playerKey .. 'link'] or Variables.varDefault(map[playerKey] .. '_page') or map[playerKey] or 'tbd'
```

Comment:
```suggestion
				map[playerKey .. 'link'] or Variables.varDefault(map[playerKey] .. '_page') or map[playerKey]
```
It can never reach the `or 'tbd'` as far as I can tell, either the first or the third will always be set if we get here